build/configs/imxrt1050-evk/imxrt1050-evk_download.sh

```diff
@@ -64,13 +64,17 @@ APP2_IMG1=${BIN_PATH}/wifi
 APP2_IMG2=${BIN_PATH}/wifi
 FLASH_IMG=${BIN_PATH}/tinyara_flash.bin
 
-# Set the flash partition sizes. This value needs to be modified whenever
-# there is a change in flash partition size in the project.
-
-let OS_PART=128*1024
-let USER_PART=128*1024
-let APP1_PART=128*1024
-let APP2_PART=256*1024
+# Extract partition sizes form CONFIG_FLASH_PART_SIZE
+OIFS=$IFS
+IFS=","
+SIZES=($CONFIG_FLASH_PART_SIZE)
+IFS=$OIFS
+
+# Set the flash partition sizes.
+let OS_PART=${SIZES[0]}*1024
+let USER_PART=${SIZES[1]}*1024
+let APP1_PART=${SIZES[2]}*1024
+let APP2_PART=${SIZES[4]}*1024
```

Comment:
This is too dedicated to loadable_elf_apps. Is it possible searching the CONFIG_FLASH_PART_NAME to find the index of APP1_IMG1 and APP2_IMG1?

Reply:
Ok.. I changed it.